### PR TITLE
[WIP] bibcheck: rm tag / field when last subfield is deleted

### PR DESCRIPTION
**Pull request by ksachs**
- update delete_field
  to delete field when last subfield is deleted
  to delete tag when last field is deleted
  
  equivalent to bibrecord  record_delete_fields

Signed-off-by: Kirsten Sachs kirsten.sachs@desy.de

### CODE REVIEW

**Comment by ksachs** [comment]
Just a proposal.

In response to Thorstens comment on https://github.com/inspirehep/inspire/pull/168
it might be better to deal with it in bibcheck_task.py

**Comment by tsgit** [comment]
LGTM

works as advertised in my testing :+1: 

it doesn't solve the problem of one empty tag among repeatable tags, though, e.g. if there is an empty 700 (we do have some of those) and another 700 with content that is deleted. That's a separate cleanup task.

e.g.

```
record.get('700')

[([], ' ', ' ', '', 18),
 ([('a', 'Jones')], ' ', ' ', '', 19),
 ([('a', 'Smith')], ' ', ' ', '', 20)]

for pos, val in record.iterfields(['700__%']):
    print pos, val
('700__a', 1, 0) Jones
('700__a', 2, 0) Smith
```

now when I delete Jones and Smith there's still 

```
record.get('700')
[([], ' ', ' ', '', 18)]
```

left 

@kaplun what do you think?

File: modules/bibcheck/lib/bibcheck_task.py [hunk 1]
```diff
@@ -441,11 +441,18 @@ def delete_field(self, position, message=""):
...
-        self.set_amended(message)
         if position[2] is None:
-            return self._query(position[:1] + (None, None)).pop(position[1])
+            value = self._query(position[:1] + (None, None)).pop(position[1])
         else:
-            return self._query(position[:2] + (None,))[0].pop(position[2])
+            field = self._query(position[:2] + (None,))[0]
+            value = field.pop(position[2])
+            if not field:
+                self._query(position[:1] + (None, None)).pop(position[1])
```
Comment (tsgit) [inline modules/bibcheck/lib/bibcheck_task.py]: why use position[:1] when position[0] is clearer and faster?

```
In [30]: pos
Out[30]: ('700__a', 1, 2)

In [31]: %timeit pos[:1]
10000000 loops, best of 3: 36.1 ns per loop

In [32]: %timeit pos[:1]
10000000 loops, best of 3: 36.1 ns per loop

In [33]: pos
Out[33]: ('700__a', 1, 2)

In [34]: %timeit pos[0]
10000000 loops, best of 3: 25.2 ns per loop

In [35]: %timeit pos[0]
10000000 loops, best of 3: 25.4 ns per loop
```

**Comment by kaplun** [comment]
@tsgit I think it's OK assuming we then have the regular task that clean up all empty fields. Although we need to ensure is run after this one.

**Pull request merged**